src/components/code-snippet/code-snippet.ts

```diff
@@ -0,0 +1,270 @@
+    ({ showFeedback = false }: { showFeedback?: boolean }) => {
+      this._showCopyButtonFeedback = showFeedback;
+      this.requestUpdate();
+    }
+  );
+
+  /**
+   * Handles `click` event on the expando.
+   */
+  private _handleClickExpando() {
+    this._expanded = !this._expanded;
+    this.requestUpdate();
+  }
+
+  /**
+   * Handles change in slot content to determine if the content
+   */
+  private _handleSlotChange() {
+    const { type, _preNode: preNode } = this;
+    if (type === CODE_SNIPPET_TYPE.MULTI) {
+      if (preNode.getBoundingClientRect().height > 255) {
+        this._showExpando = true;
+        this.requestUpdate();
+      }
+    }
+  }
+
+  /**
+   * The `<pre>` element in the shadow DOM.
+   */
+  @query('pre')
+  private _preNode!: HTMLPreElement;
+
+  /**
+   * An assistive text for screen reader to advice a DOM node is for code snippet. Corresponds to `code-assistive-text` attribute.
+   */
+  @property({ attribute: 'code-assistive-text' })
+  codeAssistiveText = 'code-snippet';
+
+  /**
+   * The context content for the collapse button. Corresponds to `collapse-button-text` attribute.
+   */
+  @property({ attribute: 'collapse-button-text' })
+  collapseButtonText = 'Show less';
+
+  /**
+   * An assistive text for screen reader to announce, telling that the button copies the content to the clipboard.
+   * Corresponds to `copy-button-assistive-text` attribute.
+   */
+  @property({ attribute: 'copy-button-assistive-text' })
+  copyButtonAssistiveText = 'Copy to clipboard';
+
+  /**
+   * The feedback text for the copy button. Corresponds to `copy-button-feedback-text` attribute.
+   */
+  @property({ attribute: 'copy-button-feedback-text' })
+  copyButtonFeedbackText = 'Copied!';
+
+  /**
+   * The number in milliseconds to determine how long the tooltip for the copy button should remain.
+   * Corresponds to `copy-button-feedback-timeout` attribute.
+   */
+  @property({ type: Number, attribute: 'copy-button-feedback-timeout' })
+  copyButtonFeedbackTimeout = 2000;
+
+  /**
+   * The context content for the expand button. Corresponds to `expand-button-text` attribute.
+   */
+  @property({ attribute: 'expand-button-text' })
+  expandButtonText = 'Show more';
+
+  /**
+   * The type of code snippet. Corresponds to the attribute with the same name.
+   */
+  @property({ reflect: true })
+  type = CODE_SNIPPET_TYPE.SINGLE;
+
+  render() {
+    const {
+      codeAssistiveText,
+      collapseButtonText,
+      copyButtonAssistiveText,
+      copyButtonFeedbackText,
+      expandButtonText,
+      type,
+      _expanded: expanded,
+      _showCopyButtonFeedback: showCopyButtonFeedback,
+      _showExpando: showExpando,
+      _handleClickCopyButton: handleClickCopyButton,
+      _handleClickExpando: handleClickExpando,
+      _handleSlotChange: handleSlotChange,
+    } = this;
+    // Ensures no extra whitespace text node
+    // prettier-ignore
+    return html`
+      ${type === CODE_SNIPPET_TYPE.INLINE
```

Comment:
Would it maybe make sense to instead use multiple `return`s instead of bundling everything into one `html` template?

Reply:
Makes sense, thank you for your feedback! - Updated.